app/models/user_batch.rb

```diff
@@ -35,79 +36,247 @@ def create_users(mission)
     end
 
     # parse the input file as a spreadsheet
-    data = Roo::Spreadsheet.open(file)
+    @data = Roo::Spreadsheet.open(file)
 
+    validate_headers
+
+    if !@validation_error
+
+      # create the users in a transaction in case of validation error
+      User.transaction do
+
+        row_start = @data.first_row
+        @import_num = last_import_num_on_users
+
+        user_batch_attributes = parse_rows(row_start)
+
+        (0..number_of_iterations).each do |i|
+          current_attributes_batch = user_batch_attributes[row_start-1, BATCH_SIZE]
+
+          users_batch = create_users_instances(current_attributes_batch, mission)
+          create_hash_table_with_fields_and_indexes(users_batch)
+
+          validate_users_batch(users_batch)
+
+          check_uniqueness_on_objects(users_batch, 'email')
+          check_uniqueness_on_objects(users_batch, 'login')
+          check_uniqueness_on_objects(users_batch, 'phone')
+          check_uniqueness_on_objects(users_batch, 'phone2')
+
+          check_uniqueness_on_db(users_batch, 'users', 'email', row_start)
+          check_uniqueness_on_db(users_batch, 'users', 'login', row_start)
+          check_uniqueness_on_db(users_batch, 'users', 'phone', row_start, ['phone','phone2'])
```

Comment:
Seems like this should be doable with one call to the method passing phone and phone2?